docs/resources/connection_aws.md

```diff
@@ -0,0 +1,122 @@
+
+<!-- schema generated by tfplugindocs -->
+## Schema
+
+### Required
+
+- `name` (String) The identifier for the connection.
+
+### Optional
+
+- `access_key_id` (Block List, Max: 1) The access key ID to connect with.. Can be supplied as either free text using `text` or reference to a secret object using `secret`. (see [below for nested schema](#nestedblock--access_key_id))
+- `assume_role_arn` (String) The Amazon Resource Name (ARN) of the IAM role to assume.
+- `assume_role_session_name` (String) The session name to use when assuming the role.
+- `comment` (String) **Public Preview** Comment on an object in the database.
+- `database_name` (String) The identifier for the connection database in Materialize. Defaults to `MZ_DATABASE` environment variable if set or `materialize` if environment variable is not set.
+- `endpoint` (String) Override the default AWS endpoint URL.
+- `ownership_role` (String) The owernship role of the object.
+- `region` (String) The AWS region to connect to.
+- `schema_name` (String) The identifier for the connection schema in Materialize. Defaults to `public`.
+- `secret_access_key` (Block List, Max: 1) The secret access key corresponding to the specified access key ID. (see [below for nested schema](#nestedblock--secret_access_key))
+- `session_token` (Block List, Max: 1) The session token corresponding to the specified access key ID.. Can be supplied as either free text using `text` or reference to a secret object using `secret`. (see [below for nested schema](#nestedblock--session_token))
+- `validate` (Boolean) **Private Preview** If the connection should wait for validation.
+
+### Read-Only
+
+- `id` (String) The ID of this resource.
+- `qualified_sql_name` (String) The fully qualified name of the connection.
+
+<a id="nestedblock--access_key_id"></a>
+### Nested Schema for `access_key_id`
+
+Optional:
+
+- `secret` (Block List, Max: 1) The `access_key_id` secret value. Conflicts with `text` within this block. (see [below for nested schema](#nestedblock--access_key_id--secret))
+- `text` (String, Sensitive) The `access_key_id` text value. Conflicts with `secret` within this block
+
+<a id="nestedblock--access_key_id--secret"></a>
+### Nested Schema for `access_key_id.secret`
+
+Required:
+
+- `name` (String) The access_key_id name.
+
+Optional:
+
+- `database_name` (String) The access_key_id database name. Defaults to `MZ_DATABASE` environment variable if set or `materialize` if environment variable is not set.
+- `schema_name` (String) The access_key_id schema name. Defaults to `public`.
+
+
+
+<a id="nestedblock--secret_access_key"></a>
+### Nested Schema for `secret_access_key`
+
+Required:
+
+- `name` (String) The secret_access_key name.
+
+Optional:
+
+- `database_name` (String) The secret_access_key database name. Defaults to `MZ_DATABASE` environment variable if set or `materialize` if environment variable is not set.
+- `schema_name` (String) The secret_access_key schema name. Defaults to `public`.
```

Comment:
I don't quite follow, why does the `secret_access_key` have separate database and schema settings?

Reply:
Oh I see, because the secret is a separate object. Makes sense.

Reply:
Yep exactly!